src/new/sendTransaction.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import EventEmitter from 'events';
+import { MIN_POLL } from '../constants';
+import transaction from '../transaction';
+import txApi from '../api/txApi';
+import txMiningApi from '../api/txMining';
+import { AddressError, OutputValueError, ConstantNotSet, MaximumNumberOutputsError, MaximumNumberInputsError } from '../errors';
+
+/**
+ * This is transaction mining class responsible for:
+ *
+ * - Submit a job to be mined;
+ * - Update mining time estimation from time to time;
+ * - Get back mining response;
+ * - Push tx to the network;
+ *
+ * It emits the following events:
+ * 'job-submitted': after job was submitted;
+ * 'estimation-updated': after getting the job status;
+ * 'job-done': after job is finished;
+ * 'send-success': after push tx succeeds;
+ * 'send-error': if an error happens;
+ **/
+class SendTransaction extends EventEmitter {
```

Comment:
Nice class. We may have a promise that is solved after the push tx. So, users might choose the best approach for their specific case.

Reply:
Interesting, I will add it.

Reply:
Added